### ISSUE TITLE
Should not use admin-only `subscribe_trigger`

### ISSUE THREAD

**Issue by dermotduffy** [dermotduffy/frigate-hass-card#1494]
I got the same issue recently on all my non-admin users, so I dig a bit into the code. I found out that this recent change: https://github.com/dermotduffy/frigate-hass-card/commit/88bea0ae7d761d588ca00795ba40d6d0807abf55 bring in "subscribe_trigger" message type, which is limited to only admin users. 

This restriction has been discussed in https://github.com/home-assistant/android/issues/3313 and concluded as by design, and the Android app made a small change to workaround it https://github.com/home-assistant/android/pull/3417.

I tried to use this snippet below to replace current subscribeToTrigger function and it seems to work properly for my usage, but I guess further optimization would be needed to avoid multiple subscriptions. We can also try to mimic the Android app's fix and check if the user is admin first and fallback to this only for non-admin users.

```ts
export const subscribeToTrigger = async (
  hass: HomeAssistant,
  callback: SubscriptionCallback,
  options?: {
    entityID?: string | string[];
    platform?: string;
    topic?: string;
    payload?: string;
    valueTemplate?: string;
    stateOnly?: boolean;
  },
): Promise<SubscriptionUnsubscribe> => {
  return await hass.connection.subscribeMessage<{ entityId: string, newState: null | unknown }>((data) => {
    if (!!data.newState && (!options?.entityID 
      || (options.entityID instanceof String && data.entityId === options) 
      || (options.entityID instanceof Array && options.entityID.includes(data.entityId)))) {
      callback({
        ...data,
        toState: data.newState
      });
    }
  }, {
    type: 'subscribe_events',
    event_type: 'state_changed'
  })
};
```

_Originally posted by @nguyenquyhy in https://github.com/dermotduffy/frigate-hass-card/issues/1426#issuecomment-2285408453_

**Comment by dermotduffy** [comment]
Sadly, `subscribe_trigger` was really handy, and replacing it is not simple. There are two ways the card uses this:

 - Subscribing to entity state. I plan to implement my own state watcher as the card is already being passed the full HA state on every update. The alternatves are (0) use a `state_changed` event, with the downside that I cannot filter which entities to subscribe to (so all state changes get sent down the websocket), or use ` subscribe_entity` call but that gets called for all  entity changes not just entity state changes so further processing is necessary anyway.
 - Subscribing to Frigate events. Sadly, since I cannot subscribe to the MQTT topic anymore, I'll need to modify the integration to copy the events from the MQTT stream into a separate events stream that the card can subscribe to without admin privileges.

**Comment by dermotduffy** [comment]
> Subscribing to Frigate events. Sadly, since I cannot subscribe to the MQTT topic anymore, I'll need to modify the integration to copy the events from the MQTT stream into a separate events stream that the card can subscribe to without admin privileges.

It's actually even worse than this. I can't subscribe to the triggers, but access to the events stream is _also_ blocked for non-admin accounts except for a shortlist of allowed event streams from HA core (i.e. no integrations, and certainly not custom integrations).

**Comment by dermotduffy** [comment]
Fix requires a Frigate integration >`5.4.0` .
